src/cow-react/modules/trade/hooks/useTradeNavigate.ts

```diff
@@ -12,22 +14,27 @@ interface UseTradeNavigateCallback {
 export function useTradeNavigate(): UseTradeNavigateCallback {
   const history = useHistory()
   const tradeTypeInfo = useTradeTypeInfo()
+  const { chainId: currentChainId } = useWeb3React()
 
   return useCallback(
     (chainId: SupportedChainId | null | undefined, { inputCurrencyId, outputCurrencyId }: TradeCurrenciesIds) => {
       if (!tradeTypeInfo) return
 
+      const isNetworkSupported = isSupportedChainId(currentChainId)
+      // Currencies ids shouldn't be displayed in the URL when user selected unsupported network
+      const fixCurrencyId = (currencyId: string | null) => (isNetworkSupported ? currencyId || undefined : undefined)
```

Comment:
Nitpick: can this be defined outside of the hook?

Reply:
Fixed